test/examples/nodejs_postgresql_test.go

```diff
@@ -0,0 +1,404 @@
+	require.Equal(t, appNS, actSBRResponse.Metadata.Namespace, "SBR Namespace is not matched, As expected namespace is %d and actual namespace is %d\n", appNS, actSBRResponse.Metadata.Namespace)
+	require.Equal(t, sbr, actSBRResponse.Metadata.Name, "SBR Name is not matched, As expected name is %d and actual name is %d\n", sbr, actSBRResponse.Metadata.Name)
+	require.Equal(t, expResource, actSBRResponse.Spec.ApplicationSelector.Resource, "SBR application resource is not matched, As expected application resource is %d and actual application resource is %d\n", expResource, actSBRResponse.Spec.ApplicationSelector.Resource)
+	require.Equal(t, appName, actSBRResponse.Spec.ApplicationSelector.ResourceRef, "SBR application resource ref is not matched, As expected application resource ref is %d and actual application resource ref  is %d\n", appName, actSBRResponse.Spec.ApplicationSelector.ResourceRef)
+	require.Equal(t, expKind, actSBRResponse.Spec.BackingServiceSelector.Kind, "SBR BackingServiceSelector kind is not matched, As expected BackingServiceSelector kind is %d and actual BackingServiceSelector kind  is %d\n", expKind, actSBRResponse.Spec.BackingServiceSelector.Kind)
+	require.Equal(t, dbName, actSBRResponse.Spec.BackingServiceSelector.ResourceRef, "SBR BackingServiceSelector ResourceRef is not matched, As expected BackingServiceSelector ResourceRef is %d and actual BackingServiceSelector ResourceRef  is %d\n", dbName, actSBRResponse.Spec.BackingServiceSelector.ResourceRef)
+
+	t.Log(" Fetching the route of an app ")
+	route := util.GetCmdResult("", "oc", "get", "route", appName, "-n", appNS, "-o", `jsonpath={.status.ingress[0].host}`)
+	t.Logf("-> ROUTE - %s \n", route)
+
+	host := "http://" + route
+	checkNodeJSAppFrontend(t, host, dbName)
+
+	t.Log("Fetching the details of secret")
+	secret := util.GetCmdResult("", "oc", "get", "sbr", sbr, "-n", appNS, "-o", `jsonpath={.status.secret}`)
+	require.Contains(t, secret, sbr, "service binding secret detail -> %d \n", secret)
+	t.Logf("-> service binding secret detail -> %s \n", secret)
+
+	t.Log("Fetching the env details of binding")
+	env := util.GetCmdResult("", "oc", "get", "deploy", appName, "-n", appNS, "-o", `jsonpath={.spec.template.spec.containers[0].env}`)
+	require.Contains(t, env, "ServiceBindingOperator", "service binding env detail -> %d \n", env)
+	t.Logf("-> service binding env detail ->%s \n", env)
+
+	t.Log("Fetching the envFrom details of binding")
+	envFrom := util.GetCmdResult("", "oc", "get", "deploy", appName, "-n", appNS, "-o", `jsonpath={.spec.template.spec.containers[0].envFrom}`)
+	require.Contains(t, envFrom, sbr, "service binding envFrom detail -> %d \n", envFrom)
+	t.Logf("-> service binding envFrom detail -> %s \n", envFrom)
+}
+
+func checkClusterAvailable(t *testing.T) {
+	if !clusterAvailable {
+		t.Skip("Cluster is not available, skipping")
+	}
+}
+
+func checkNodeJSAppFrontend(t *testing.T, startURL string, expectedHeader string) {
+	wd, svc := initSelenium(t)
```

Comment:
Instead of Selenium WebDriver, can we use Go (`net/http` package) to test the content of the pages?

Reply:
@baijum We have considered that but the problem is that the page and the content that we verify (the name of the DB - value of the `dbNameCM` variable) is rendered by JavaScript. If you use plain HTTP (such as `net/http` or `curl`) to get the page's content, the only thing you get is plain non-rendered HTML code with JavaScript variables `<h1>Node.js Crud Application (DB: {{ dbNameCM }})</h1>`. To get the rendered page we need something to render the page properly and that is what the `chromium` does and we can than use the WebDriver to get that.
If there's a "only-go" way to do the same, that would be ideal, but we are not aware of any such package.

Reply:
@pmacik The example apps are also created by us, right? Then why we can't have an API endpoint that gives all details required for verification?

Reply:
@baijum That is a very good point - the application actually already exposes an API endpoint to provide that DB name (the one that the web page calls to render the header), so we will use that instead of selenium at all. Thanks for that insight!